### PR TITLE
mysql 8.0.25

### PR DESCRIPTION
**Pull request by yahonda**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by yahonda** [comment]
Let me take a look at CI tests failures.

**Comment by carlocab** [comment]
Looks to be all CMake errors. You can have a closer look at the logs with
```
gh run download -RHomebrew/homebrew-core 847981941
```

**Comment by cho-m** [comment]
Switching to "system" over "bundled" (vendored) dependencies can help. 
At least using macOS's `zlib` and Homebrew's `zstd` will deal with CMake errors causing failure:
```make
-- ZLIB_VERSION (bundled) is 1.2.11
CMake Error at extra/zstd/CMakeLists.txt:54 (TARGET_COMPILE_OPTIONS):
  Cannot specify compile options for target "zstd_objlib" which is not built
  by this project.


-- ZSTD_LEGACY_SUPPORT not defined!
-- ZSTD_VERSION (bundled) is 1.4.3
```

The logs also show `libevent` as bundled, though I don't think it will cause build to fail either way.

---

The following is a link to all dependencies that can be switched to "system".
Some aren't used in Formula and can potentially be ignored
(e.g. current formula doesn't use `curl`. in logs: `WITH_CURL=none, not using any curl library.`)

https://github.com/mysql/mysql-server/blob/mysql-8.0.25/CMakeLists.txt#L1268-L1278

**Comment by yahonda** [comment]
Thanks for the information.

Taking a look at the log file when it installs MySQL 8.0.23, there is no error around zlib and zstd as follows then your suggestion looks correct.

- /Users/yahonda/Library/Logs/Homebrew/mysql/01.cmake when it installs MySQL 8.0.23

```
-- ZLIB_VERSION (bundled) is 1.2.11
-- ZSTD_LEGACY_SUPPORT not defined!
-- ZSTD_VERSION (bundled) is 1.4.3
```

> Switching to "system" over "bundled" (vendored) dependencies can help.

Since I'm not familiar with Homebrew yet, I do not know how to switch bundled dependencies to system one. Would you give me some advice? 

Thanks.

**Comment by SMillerDev** [comment]
> Since I'm not familiar with Homebrew yet, I do not know how to switch bundled dependencies to system one.

This isn't a homebrew setting. It would be up to the upstream build system to make this switch.

**Comment by carlocab** [comment]
Switching from a bundled to a Homebrew dependency usually amounts to adding the right `depends_on` and then passing the correct build flag.

**Comment by yahonda** [comment]
Based on the message "  Cannot specify compile options for target "zstd_objlib" which is not built
  by this project." it may be related to https://github.com/mysql/mysql-server/commit/52a5242d4cf1

This commit has been merged since `mysql-8.0.24` tag then it makes sense why this pull request and https://github.com/Homebrew/homebrew-core/pull/75633 fail to build.

**Comment by cho-m** [comment]
As Homebrew prefers to avoid vendored dependencies, it should be good to switch to system dependencies whether or not the build failed.

MySQL's CMakeLists.txt file shows the available `OPTION`s. The existing flags in this formula (e.g. protobuf) and how other formulae set up dependencies can be used as reference.

You may need to check if any dependencies are build-only, but a basic modification to use system alternatives to the 3 previously mentioned bundled dependencies is:
```diff
@@ -19,10 +18,13 @@ class Mysql < Formula
   end

   depends_on "cmake" => :build
+  depends_on "libevent"
   depends_on "openssl@1.1"
   depends_on "protobuf"
+  depends_on "zstd"

   uses_from_macos "libedit"
+  uses_from_macos "zlib"

   conflicts_with "mariadb", "percona-server",
     because: "mysql, mariadb, and percona install the same binaries"
@@ -48,6 +50,9 @@ class Mysql < Formula
       -DWITH_EDITLINE=system
       -DWITH_SSL=#{Formula["openssl@1.1"].opt_prefix}
       -DWITH_PROTOBUF=system
+      -DWITH_LIBEVENT=system
+      -DWITH_ZLIB=system
+      -DWITH_ZSTD=system
       -DWITH_UNIT_TESTS=OFF
       -DENABLED_LOCAL_INFILE=1
       -DWITH_INNODB_MEMCACHED=ON
```

CMake will then be pull in the system versions,
```make
-- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib/libz.tbd (found version "1.2.11") 
-- ZLIB_VERSION (system) is 1.2.11
-- ZSTD_VERSION (system) is 1.5.0
-- The openssl command does not support zlib
...
-- LIBEVENT_INCLUDE_DIRS /opt/homebrew/include
-- LIBEVENT_VERSION_STRING 2.1.12-stable
-- LIBEVENT_VERSION (system) 2.1.12
```

---

Whether this should be done for the other dependencies is still a question.
e.g. output for `icu`
```
-- ICU_INCLUDE_DIRS /tmp/mysql-20210518-39313-1wj4d5j/mysql-8.0.25/extra/icu/source/common;/tmp/mysql-20210518-39313-1wj4d5j/mysql-8.0.25/extra/icu/source/stubdata;/tmp/mysql-20210518-39313-1wj4d5j/mysql-8.0.25/extra/icu/source/i18n
```

File: Formula/mysql.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class Mysql < Formula
   desc "Open source relational database management system"
   homepage "https://dev.mysql.com/doc/refman/8.0/en/"
-  url "https://cdn.mysql.com/Downloads/MySQL-8.0/mysql-boost-8.0.23.tar.gz"
-  sha256 "1c7a424303c134758e59607a0b3172e43a21a27ff08e8c88c2439ffd4fc724a5"
+  url "https://cdn.mysql.com/Downloads/MySQL-8.0/mysql-boost-8.0.25.tar.gz"
+  sha256 "93c5f57cbd69573a8d9798725edec52e92830f70c398a1afaaea2227db331728"
   license "GPL-2.0-only"
-  revision 2
```
Comment (yahonda) [inline Formula/mysql.rb:7]: I'm wondering if it needs to specify `revision1 1` explicitly.

**Comment by yahonda** [comment]
Thanks for the useful comment @cho-m. I have added one more commit and rebased it from the master branch then force-pushed.

File: Formula/mysql.rb (hunk 1 above)
Comment (carlocab) [reply Formula/mysql.rb:7]: `revision` starts at `0`, and we omit it whenever it's `0`.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
We probably want to use brewed `lz4` and `icu` here. But I'll do that in a follow-up PR. (Or someone else can take that too; that's fine with me.)

**Pull request closed without merging**